src/index.js

```diff
@@ -13,45 +13,94 @@
  * @returns {typeof EPromise} EPromise class
  */
 export default function makeEPromiseClass(BasePromise) {
-  // A remoteRelay must additionally have an AWAIT_FAR method
-  const localRelay = {
-    GET(p, key) {
-      return p.then(o => o[key]);
-    },
-    PUT(p, key, val) {
-      return p.then(o => (o[key] = val));
-    },
-    DELETE(p, key) {
-      return p.then(o => delete o[key]);
-    },
-    POST(p, optKey, args) {
-      if (optKey === undefined || optKey === null) {
-        return p.then(o => o(...args));
-      }
-      return p.then(o => o[optKey](...args));
-    },
-  };
-
-  const relayToPromise = new WeakMap();
+  const presenceToResolvedRelay = new WeakMap();
   const promiseToRelay = new WeakMap();
 
+  // This special relay accepts Promises, and forwards
+  // the resolved Presence to its corresponding resolvedRelay.
+  //
+  // If passed a Promise that is not a Presence, perform
```

Comment:
This comment does not make sense to me. A promise can be to a presence, but a promise never is a presence.

Reply:
Fixed in next commit.